frontend/javascripts/oxalis/view/node_context_menu.js

```diff
@@ -181,9 +201,62 @@ function NoNodeContextMenuOptions({
   globalPosition,
   viewport,
   createTree,
+  segmentationLayer,
+  dataset,
+  createMesh,
+  zoomStep,
 }: NoNodeContextMenuProps) {
+  const [availableMeshFiles, setAvailableMeshFiles] = useState([]);
+
+  const updateAvailableMeshFiles = async () => {
+    if (segmentationLayer) {
+      const layerName = segmentationLayer.fallbackLayer || segmentationLayer.name;
+      setAvailableMeshFiles(
+        await getMeshfilesForDatasetLayer(dataset.dataStore.url, dataset, layerName),
+      );
+    }
+  };
+
+  const loadMeshFromFile = async meshFileItem => {
+    const layer = Model.getSegmentationLayer();
+    if (!layer) {
+      throw new Error("No segmentation layer found");
+    }
+    const segmentationCube = layer.cube;
+    const fileName = meshFileItem.key;
+    const id = segmentationCube.getDataValue(globalPosition, null, zoomStep);
+    if (segmentationLayer == null) return;
+    const layerName = segmentationLayer.fallbackLayer || segmentationLayer.name;
+    const availableChunks = await getMeshfileChunksForSegment(
+      dataset.dataStore.url,
+      dataset,
+      layerName,
+      fileName,
+      id,
+    );
+    for (const chunkPos of availableChunks) {
+      // eslint-disable-next-line no-await-in-loop
+      const stlData = await getMeshfileChunkData(
+        dataset.dataStore.url,
+        dataset,
+        layerName,
+        fileName,
+        id,
+        chunkPos,
+      );
+      const geometry = parseStlBuffer(stlData);
+      getSceneController().addIsosurfaceFromGeometry(geometry, id);
+    }
+    createMesh(id, globalPosition);
+  };
```

Comment:
this code should be in the isosurface saga imo. i think, there is also some duplication with code in the mesh-view itself. both places should be unified and be moved into the saga (or another module).